### PR TITLE
Add deprecation notice around Truffle support

### PR DESCRIPTION
**Pull request by aefhm**
## Description

Consensys is [sunsetting](https://consensys.io/blog/consensys-announces-the-sunset-of-truffle-and-ganache-and-new-hardhat) Truffle.

We have already broadcasted this news in [Discord](https://discord.com/channels/748635004384313474/960599114993762304/1155489141996392588), but need to update our documentation to reflect this message.

**_NOTE_** that the porting over an existing Eth app is not explicitly covered now.

## TODO
- [x] Update video link in Sapphire to this one which uses Hardhat only https://www.youtube.com/watch?v=LDLz06X_KNY

### CODE REVIEW

**Comment by matevz** [comment]
I would devote an hour or two, read through https://docs.oasis.io/dapp/sapphire/quickstart and https://docs.oasis.io/dapp/emerald/writing-dapps-on-emerald#create-dapp-on-emerald-with-truffle and completely remove the truffle-related sections. And then put admonition something like

```
::: info Truffle support

Oasis removed support for Truffle on 2023-10-01. Developers are encouraged to migrate to Hardhat instead.
The original Truffle tutorial is available at https://github.com/oasisprotocol/docs/blob/2f4a1a3c217b82687ab9440bf051762ae369ed45/docs/dapp/sapphire/quickstart.mdx and the deprecated Truffle
example at https://github.com/oasisprotocol/sapphire-paratime/tree/main/examples/truffle.

:::
```

Afterwards, https://github.com/oasisprotocol/sapphire-paratime/pull/191 can be merged, a new version of ts-client released and examples/truffle removed.

**Review by matevz — CHANGES_REQUESTED** [review]
See https://github.com/oasisprotocol/docs/pull/547#issuecomment-1736861937

File: docs/dapp/sapphire/quickstart.mdx [hunk 1]
```diff
@@ -4,275 +4,34 @@ import {findSidebarItem} from '@site/src/sidebarUtils';
...
-       network_id: 0x5aff,
-     },
-   },
-```
-
-This `wrap` function takes any kind of provider or signer you've got and turn
-it into one that works with Sapphire and confidentiality.
-For the most part, wrapping your signer/provider is the most you'll need to do
-to get your dApp running on Sapphire, but that's not a complete story since an
-unmodified contract may leak state through normal operation.
```
Comment (aefhm) [inline docs/dapp/sapphire/quickstart.mdx:240]: We may potentially want to find a place to say this again.

File: docs/dapp/sapphire/quickstart.mdx [hunk 2]
```diff
@@ -4,275 +4,34 @@ import {findSidebarItem} from '@site/src/sidebarUtils';
...
-been returned.
-What's happened here is that the Web3 gateway does not have the key used to
-decrypt the storage slot, so a default value is returned.
-
-Indeed, the gateway does not even have the key needed to decrypt the _key_ in
-the MKVS; it can tell _that_ a storage slot was written, but not which one
-(although it can make a very good guess by reading the contract code).
-
-All in all, you can see that confidentiality is in effect, but it's not
-something end-users need to think too much about.
```
Comment (aefhm) [inline docs/dapp/sapphire/quickstart.mdx:268]: This is probably the most important point in the Truffle example.

File: docs/dapp/sapphire/quickstart.mdx [hunk 3]
```diff
@@ -336,6 +95,12 @@ any network config entry corresponding to the Sapphire's chain ID will
 automatically be wrapped with Sapphire specifics for encrypting and signing the
 transactions.
 
+#### Get some Sapphire Testnet tokens
+
+Now for the fun part. We need to configure the Sapphire network and get some tokens.
+Hit up the one and only [Oasis Testnet faucet] and this time select "Sapphire".
```
Comment (matevz) [inline docs/dapp/sapphire/quickstart.mdx]: 1. [Oasis Testnet faucet] has no link.
2. Also, can you put `any network config entry corresponding to the Sapphire's chain ID will
automatically be wrapped with Sapphire specifics for encrypting and signing the
transactions` in bold?

Comment (matevz) [inline docs/dapp/sapphire/quickstart.mdx]:
```suggestion
Hit up the one and only [Oasis Testnet faucet] and select "Sapphire".
```

File: docs/dapp/sapphire/quickstart.mdx [hunk 4]
```diff
@@ -414,9 +179,8 @@ Best of luck on your future forays into confidentiality!
 
 :::info Example
 
-Visit the Sapphire ParaTime repository to download the final [Truffle]
-[truffle-example] and [Hardhat][hardhat-example] examples of this
-quickstart.
+Visit the Sapphire ParaTime repository to download the [Hardhat][hardhat-example]
+example of this quickstart.
```
Comment (matevz) [inline docs/dapp/sapphire/quickstart.mdx:186]:
```suggestion

:::

:::info Example

If your project will also have a web frontend, we suggest you start with the
official [Oasis starter] files.

:::

[Oasis starter]: https://github.com/oasisprotocol/demo-starter

```

**Review by matevz — APPROVED** [review]

File: docs/dapp/sapphire/quickstart.mdx (hunk 3 above)
Comment (aefhm) [reply docs/dapp/sapphire/quickstart.mdx]: oops 🤐 hvala

**Pull request merged**